packages/types/src/reef-knot-wallets-modal.ts

```diff
@@ -2,35 +2,23 @@ import type { ComponentType } from 'react';
 import type { ModalProps } from '@reef-knot/ui-react';
 import type { ConnectWalletButtonProps } from './wallet-connect-button';
 
-export type MetricsProp<WalletIdsList extends string = string> = {
-  events?: {
-    connect?: {
-      handlers: Partial<Record<WalletIdsList, () => void>>;
-    };
-    click?: {
-      handlers: Partial<
-        Record<
-          WalletIdsList | 'termsAccept' | 'walletsMore' | 'walletsLess',
-          () => void
-        >
-      >;
-    };
-  };
-};
-
 export type ButtonComponentsByConnectorId = {
   [key: string]: ComponentType<ConnectWalletButtonProps>;
 };
 
 export type ReefKnotWalletsModalConfig<I extends string = string> = {
   buttonComponentsByConnectorId: ButtonComponentsByConnectorId;
-  metrics: MetricsProp<I>;
   walletsShown: I[];
   walletsPinned: I[];
   walletsDisplayInitialCount?: number;
   linkTerms?: string;
   linkPrivacyNotice?: string;
   linkDontHaveWallet?: string;
+  onConnectStart?: (walletId: I) => void;
+  onConnectSuccess?: (walletId: I) => void;
+  onClickTermsAccept?: (isAccepted: boolean) => void;
```

Comment:
Maybe object arg shape is better, because it would allows us to add more info to callbacks with backwards comp.

Reply:
Sure, it could be convenient in the future